### ISSUE TITLE
ExpressiveAnnotations core assembly not added project files

### ISSUE THREAD

**Issue by jschroed** [fsprojects/Paket#1466]
I started having an issue adding ExpressiveAnnotations core assembly to my projects. It looks like the problem started in version 2.44.5. (Version 2.44.4 works fine.)

I have "nuget ExpressiveAnnotations.dll = 2.3.6" in my paket.dependencies file and "ExpressiveAnnotations.dll" in my paket.references. My paket.dependencies has this at the top:
```
content: none
import_targets: false
framework: net45
redirects: off
```

When I run paket update or install with version 2.44.4 of paket, ExpressiveAnnotations is added to my csproj like this:
```
  <Choose>
    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.5'">
      <ItemGroup>
        <Reference Include="ExpressiveAnnotations">
          <HintPath>..\packages\ExpressiveAnnotations.dll\lib\net45\ExpressiveAnnotations.dll</HintPath>
          <Private>True</Private>
          <Paket>True</Paket>
        </Reference>
      </ItemGroup>
    </When>
  </Choose>
```
Version 2.44.5 and higher do not add "ExpressiveAnnotations" to my csproj file. (Note: I haven't tried all versions and I haven't tried any of the 3.0.0 versions.) Is there something I need to change in my configuration or is this an issue with paket?

**Comment by forki** [comment]
The issue is a bit tricky. First of all: there is no ExpressiveAnnotations.dll = 2.3.6 - see https://www.nuget.org/packages/ExpressiveAnnotations

But I assum somehow you have that version in your local nuget cache. In Paket 2.44.4 we noticed a bug changed our cache format so that 2.44.5 and above will not read older cache files and try to get everything from scratch. And since the package version is deleted (@maartenba ist that correct?) paket restore fails.

**Comment by maartenba** [comment]
I don't see a 2.3.6. It wasn't deleted, it wasn't uploaded.

**Comment by forki** [comment]
Thx Maarten.

So where was it coming from?

**Comment by inosik** [comment]
I tried this, and it seems that the refactorings from d2676157 broke some string stuff. The `.dll` in the package name breaks the install, because the new [`String.endsWithIgnoreCase` function](https://github.com/fsprojects/Paket/blob/master/src/Paket.Core/Utils.fs#L397-L398) is faulty.

**Comment by inosik** [comment]
From fsi:
```
module String =
    let endsWithIgnoreCase (target : string) (text : string) =
        text.IndexOf (target, System.StringComparison.OrdinalIgnoreCase) >= text.Length - target.Length

> String.endsWithIgnoreCase ".dll" "packages/ExpressiveAnnotations.dll/lib/net45/ExpressiveAnnotations.dll";;
val it : bool = false
> String.endsWithIgnoreCase ".dll" "packages/ExpressiveAnnotations.dll/lib/net45/ExpressiveAnnotations.xml";;
val it : bool = false
> String.endsWithIgnoreCase ".dll" "packages/ExpressiveAnnotations/lib/net45/ExpressiveAnnotations.xml";;
val it : bool = false
> String.endsWithIgnoreCase ".dll" "packages/ExpressiveAnnotations/lib/net45/ExpressiveAnnotations.dll";;
val it : bool = true

**Comment by forki** [comment]
So there are two packages? One with .dll at the end?

**Comment by inosik** [comment]
Yes, the one with `.dll` in its id seems to be a kind of "Core" package.

**Comment by inosik** [comment]
`LastIndexOf` instead of `IndexOf` in [this line](https://github.com/fsprojects/Paket/blob/master/src/Paket.Core/Utils.fs#L398) should do the trick.

**Comment by forki** [comment]
yes. already deploying fix. thanks everybody.

**Comment by jschroed** [comment]
Thanks forki (and everyone else)! The issue is resolved for me with version 2.49.8.
